ffi-convert/src/types.rs

```diff
@@ -114,38 +116,60 @@ pub struct CArray<T> {
     size: usize,
 }
 
-impl<U: AsRust<V>, V> AsRust<Vec<V>> for CArray<U> {
+impl<U: AsRust<V> + 'static, V> AsRust<Vec<V>> for CArray<U> {
     fn as_rust(&self) -> Result<Vec<V>, AsRustError> {
         let mut vec = Vec::with_capacity(self.size);
+
+        let values = unsafe { std::slice::from_raw_parts_mut(self.data_ptr as *mut U, self.size) };
```

Comment:
I would leave this inside the if checking the size, not need to try and do stuff with `self.data_ptr` if the size is 0, this leaves the possibility for clients to leave this at null in that case, but the doc for slice specifies the pointer should be not null